### PR TITLE
Reset DataQueue if underlying database get corrupted

### PR DESCRIPTION
**Pull request by praveek**
https://github.com/adobe/aepsdk-core-android/issues/512

This PR contains changes to address potential database corruption concerns. The changes are contained inside DataQueue as to avoid any public API changes. 

With this change 
- Any operation modifying the data queue will reset the database on failure. 
- Any operation querying the data queue will continue with current behaviour by returning default values denoting failure.

### CODE REVIEW

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java [hunk 1]
```diff
@@ -349,4 +339,62 @@ private void createTableIfNotExists() {
...
+    private void resetDatabase() {
+        Log.warning(
+                ServiceConstants.LOG_TAG,
+                LOG_PREFIX,
+                "resetDatabase - Resetting database (%s) as it is corrupted",
+                databasePath);
+
+        try {
+            FileUtils.deleteFile(new File(databasePath), false);
+            createTableIfNotExists();
```
Comment (kevinlind) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java:391]: Do you need to synchronize on the `dbMutex` here?

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java [hunk 2]
```diff
@@ -349,4 +339,62 @@ private void createTableIfNotExists() {
                         "createTableIfNotExists - Error creating/accessing table (%s)  ",
                         TABLE_NAME));
     }
+
+    /**
+     * Adds a a Table with name {@link #TABLE_NAME}, if not already exists in database at path
+     * {@link #databasePath}.
+     */
```
Comment (kevinlind) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java:346]: Add a comment here stating the caller must hold the `dbMutex` when calling this function.

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java (hunk 1 above)
Comment (kevinlind) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java:391]: Ah, I see that you are. Can you add a comment to this method stating the caller must hold the `dbMutex`?

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java [hunk 3]
```diff
@@ -349,4 +339,62 @@ private void createTableIfNotExists() {
...
+                                            + " ?, ?)")) {
+                        insertStatement.bindString(INDEX_UUID, dataEntity.getUniqueIdentifier());
+                        insertStatement.bindLong(
+                                INDEX_TIMESTAMP, dataEntity.getTimestamp().getTime());
+                        insertStatement.bindString(
+                                INDEX_DATA,
+                                dataEntity.getData() != null ? dataEntity.getData() : "");
+                        long rowId = insertStatement.executeInsert();
+                        return rowId >= 0;
+                    } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/SQLiteDataQueue.java:372]: Got it. Makes sense 👍🏼

**Review by prudrabhat — APPROVED** [review]

**Review by kevinlind — APPROVED** [review]

**Comment by emdobrin** [comment]
👍 thanks for the change @praveek and the new tests added

**Pull request merged**